app/models/user_group.rb

```diff
@@ -198,7 +173,7 @@ def senior_delegate
   end
 
   def lead_role
-    self.active_roles.find { |role| role.is_lead? }
+    active_roles.includes(:group, metadata: [:status]).find { |role| role.is_lead? }
```

Comment:
Have you tried locally whether this works? Normally, loading polymorphic associations like `metadata` through an `includes` throws errors...

Reply:
Scrap that, I just tried this and surprisingly it worked :O

Reply:
Yes it worked, and as per our discussion in slack, removed :status.